WordPress/Classes/ViewRelated/Likes/LikesListController.swift

```diff
@@ -97,16 +97,38 @@ class LikesListController: NSObject {
         // shows the loading cell and prevents double refresh.
         isLoadingContent = true
 
-        fetchLikes(success: { [weak self] users in
+        let successBlock: ([LikeUser]) -> Void = { [weak self] users in
             self?.likingUsers = users
             self?.isLoadingContent = false
-        }, failure: { [weak self] _ in
+        }
+
+        let failureBlock: (Error?) -> Void = { [weak self] _ in
             // TODO: Handle error state
             self?.isLoadingContent = false
-        })
+        }
+
+        fetchStoredLikes(completion: successBlock)
+        fetchLikes(success: successBlock, failure: failureBlock)
+    }
+
+    /// Fetch Likes from Core Data depending on the notification's content type.
+    /// - Parameters:
+    ///   - completion: Closure to be called when the fetch is complete.
+    private func fetchStoredLikes(completion: @escaping ([LikeUser]) -> Void) {
+
+        let users: [LikeUser]
+
+        switch content {
+        case .post(let postID):
+            users = postService.likeUsersFor(postID: postID, siteID: siteID)
+        case .comment(let commentID):
+            users = commentService.likeUsersFor(commentID: commentID, siteID: siteID)
+        }
+
+        completion(users)
     }
```

Comment:
I'm curious about the use of the completion block rather than just returning the fetched users.  There's not really an async element to what this method and the services are doing.